### PR TITLE
Convert NWP grib files to Zarr intermediate

### LINKED ISSUE #344: Convert NWP grib files to Zarr intermediate
## Detailed Description
#122 found that it'll be way too slow to load NWPs from native files directly.  So we need to convert to Zarr first.

All of "file 3" listed in [this Met Office PDF](http://cedadocs.ceda.ac.uk/1334/1/uk_model_data_sheet_lores1.pdf) look useful for helping to predict the evolution of clouds.

Related issues:

- #29
- #30
- #44
- #122
- #196
- https://github.com/openclimatefix/Satip/issues/14

Also see:

- [Met Office's data docs](http://cedadocs.ceda.ac.uk/1334/1/uk_model_data_sheet_lores1.pdf)
- https://github.com/JackKelly/ng_nwp/blob/master/extract_grib.ipynb
- [OCF's data docs on NWP data](https://docs.google.com/document/d/1HO0W-L0hWhV925MWwGWmqkOQjKnNBykruMTf6OCwwak/edit#heading=h.tq5tgjh77cvt)
- [rechunk_nwp_data.py](https://github.com/openclimatefix/nowcasting_dataset/blob/3a3f2e652268de2348d13aa744f3772d935fef8d/scripts/rechunk_nwp_data.py)

Something like this should work:

```python
ds = xr.open_dataset(
    filename,
    engine="cfgrib",

    # Wind direction and wind speed are the only variables which are at 
    # 10 meters above ground in the UKV GRIB files.
    #backend_kwargs={'filter_by_keys': {'level': 10}}
    backend_kwargs=dict(filter_by_keys={'stepType': 'instant'})
)
```

And load twice:  Once with `stepType=instant` and one with `stepType=accum`.  And watch for all the fun exceptions that cfgrib throws :)

This looks like it'll be quite slow.  So definitely need to parallelise.  If it's super-slow then might be best to FTP new grib files to the cloud and do the processing in parallel in the cloud.

### PR DESCRIPTION
**Pull request by JackKelly**
# Pull Request

## Description

Write a script to convert NWP native files (in `grib` format) to a Zarr.

Fixes #344 

This script ended up being more complex than I was imagining!

Later, as part of #359, this script will be moved into its own repository, and the functions will be put into a nice little Python package (with unittests!)

## How Has This Been Tested?

The script runs successfully and has converted all our NWP data to Zarr.

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: scripts/convert_NWP_grib_to_zarr.py [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+    "heightAboveGroundLayer",
+    "atmosphere",
+    "cloudBase",
+    "surface",
+    "meanSea",
+    "level",
+)
+
+
+@click.command()
```
Comment (peterdudfield) [inline scripts/convert_NWP_grib_to_zarr.py:116]: nice use of click

File: scripts/convert_NWP_grib_to_zarr.py [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+2) Group the .grib filenames by the NWP initialisation datetime (which is the datetime given
+   in the grib filename).  For example, this groups together all the Wholesale1 and
+   Wholesale2 files.
+3) If the `destination_zarr_path` exists then find the last NWP init time in the Zarr,
+   and ignore all grib files with init times before that time.  This allows the script to
+   re-start where it left off if it crashes, or if new grib files are downloaded.
+4) Use multiple worker processes.  Each worker process is given a list of
+   grib files associated with a single NWP init datetime.  The worker process loads these grib
+   files and does some simple post-processing, and then the worker process appends to the
+   destination zarr.
```
Comment (peterdudfield) [inline scripts/convert_NWP_grib_to_zarr.py:41]: great notes

**Review by peterdudfield — APPROVED** [review]
For me it looks all very good. I like the docs and comments you have done. That makes it a lot easier. 

There's probably a few things like 
- tests,
- move functions to their own file
- hard coded values, move to their own file
- add a bit more logging
but I think these can all wait until #359

**Review by jacobbieker — APPROVED** [review]
LGTM! I like all the comments. Made it really easy to follow

**Pull request merged**